### PR TITLE
Update to gradle 4.4 and agp 3.1.3

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal
Updates to gradle 4.4 and AGP 3.1.3 (the minimum versions expected by Android Studio stable).

## Changeset

- Gradle 4.4
- AGP 3.1.3
- Support lib 27.1.1
- Bugsnag plugin 3.3.0

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ android:
   components:
     - tools
     - platform-tools
-    - build-tools-27.0.0
+    - build-tools-27.0.3
     - android-19
```
Comment (martin308) [inline .travis.yml:11]: I noticed that we also have `android-16` here in https://github.com/bugsnag/bugsnag-android, should that be added here as well?

**Review by martin308 — APPROVED** [review]
I didn't run this but the changes look sensible to me. I left one comment around a difference between this and the standard bugsnag-android travis setup

File: .travis.yml (hunk 1 above)
Comment (fractalwrench) [reply .travis.yml:11]: I'll create a ticket to address that separately - there may have been some reason we skipped setting up 16 originally.
